src/components/lib/sl/sl.c

```diff
@@ -445,13 +437,19 @@ sl_thd_comp_init(struct cos_defcompinfo *comp, int is_sched)
 	if (is_sched) {
 		t = sl_thd_aep_alloc_intern(NULL, NULL, comp, SL_THD_PROPERTY_OWN_TCAP | SL_THD_PROPERTY_SEND);
 	} else {
-		struct cos_aep_info *aep = cos_sched_aep_get(comp);
+		struct cos_aep_info *sa = cos_sched_aep_get(comp), *aep = NULL;
+
+		aep = sl_thd_alloc_aep_backend();
+		if (!aep) goto done;
 
+		*aep = *sa;
```

Comment:
It isn't clear to me why copying is the right answer here and above.  Copying breaks future equality comparisons between `aep`s (not sure if we care), and any future changes to the aep (again, not sure if we care).

Reply:
Got it, I'll add some comments. 
Copying works here because the threads are not created from here, they're created outside of `sl` and that the kernel upcall will not use this `cos_aep_info` data structure to look up it's `fn` and `data`.